test/Tester/ActivationServices/IStorageActivationService.cs

```diff
@@ -0,0 +1,78 @@
+﻿
+using Orleans.Runtime;
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Orleans;
+
+namespace Tester
+{
+    public interface IStorageActivationService<TState>
+    {
+        string Name { get; }
+
+        TState State { get; set; }
+
+        Task SaveAsync();
+
+        string GetExtendedInfo();
+    }
+
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public class StorageActivationServiceAttribute : Attribute
+    {
+        public string StateName { get; }
+        public string StorageProviderName { get; }
+
+        public StorageActivationServiceAttribute(string storageProviderName = null, string stateName = null)
+        {
+            this.StorageProviderName = storageProviderName;
+            this.StateName = stateName;
+        }
+    }
+
+    public class AttributedStorageActivationService<TState> : IStorageActivationService<TState>
+    {
+        private readonly IStorageActivationService<TState> storageActivationService;
+
+        public AttributedStorageActivationService(StorageActivationServiceFactory<TState> factory, IGrainActivationContext context)
+        {
+            var parameter = ActivationServiceUtilities.BindToConstructorParameter(context, typeof(IStorageActivationService<TState>));
```

Comment:
It seems that activation service must call `BindToConstructorParameters` to wire itself into the system. To achieve that, I wonder what's the thought behind using `ActivationServicesUtilities `vs `ActivationServiceBaseClass`. From the perspective of OOP, a base class make more sense to me. And also IMO it is slightly easier for user to remember and use.